matchms/Pipeline.py

```diff
@@ -86,32 +84,51 @@ def __init__(self, config_file=None, progress_bar=True):
         self.spectrums_queries = []
         self.spectrums_references = []
         self.is_symmetric = False
-        self._initialize_workflow_dict(config_file)
+        self.import_workflow_from_yaml(config_file)
         self.scores = None
         self.progress_bar = progress_bar
 
-    def _initialize_workflow_dict(self, config_file):
+    def _initialize_spectrum_processor_queries(self):
+        """Initialize spectrum processing workflow for the query spectra."""
+        self.processing_queries = initialize_spectrum_processor(
+            self.predefined_processing_queries,
+            self.additional_processing_queries
+            )
+        self.write_to_logfile(str(self.processing_queries))
+
+    def _initialize_spectrum_processor_references(self):
+        """Initialize spectrum processing workflow for the reference spectra."""
+        self.processing_references = initialize_spectrum_processor(
+            self.predefined_processing_references,
+            self.additional_processing_references
+            )
+        self.write_to_logfile(str(self.processing_references))
+
+    def import_workflow_from_yaml(self, config_file):
+        """Define Pipeline workflow based on a yaml file (config_file).
+        """
         if config_file is None:
             self.workflow = OrderedDict()
             self.workflow["importing"] = {"queries": None,
                                           "references": None}
-            self.workflow["filtering_queries"] = ["default_filters"]
-            self.workflow["filtering_refs"] = ["default_filters"]
+            self.workflow["predefined_processing_queries"] = "default"
+            self.workflow["predefined_processing_references"] = "default"
```

Comment:
@florian-huber I just realized that the yaml file will now store "default" instead of all the filter steps in default. I would remove the "predefined_processing_queries" in the yaml file and just combine the "predefined_processing_queries" and the addition_processing_queries, to make sure that changes in what we define under default will not affect the results of an analysis, when redone.

Reply:
This has been fixed in #480